flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java

```diff
@@ -45,23 +57,13 @@ public void onReceive(final Context context, Intent intent) {
       Gson gson = FlutterLocalNotificationsPlugin.buildGson();
       Type type = new TypeToken<NotificationDetails>() {}.getType();
       NotificationDetails notificationDetails = gson.fromJson(notificationDetailsJson, type);
+
       if (notificationDetails.showNotification == null || notificationDetails.showNotification) {
         FlutterLocalNotificationsPlugin.showNotification(context, notificationDetails);
       } else {
         BackgroundAlarm.start(context, notificationDetails);
       }
-      if (notificationDetails.scheduledNotificationRepeatFrequency != null) {
-        FlutterLocalNotificationsPlugin.zonedScheduleNextNotification(context, notificationDetails);
-      } else if (notificationDetails.matchDateTimeComponents != null) {
-        FlutterLocalNotificationsPlugin.zonedScheduleNextNotificationMatchingDateComponents(
-            context, notificationDetails);
-      } else if (notificationDetails.repeatInterval != null) {
-        FlutterLocalNotificationsPlugin.scheduleNextRepeatingNotification(
-            context, notificationDetails);
-      } else {
-        FlutterLocalNotificationsPlugin.removeNotificationFromCache(
-            context, notificationDetails.id);
-      }
+      FlutterLocalNotificationsPlugin.scheduleNextNotification(context, notificationDetails);
```

Comment:
There was a merge conflict here so it would be great with a double check if I did correct.